zcash_client_backend/src/proposal.rs

```diff
@@ -48,6 +48,10 @@ pub enum ProposalError {
     PaymentPoolsMismatch,
     /// The proposal tried to spend a change output. Mark the `ChangeValue` as ephemeral if this is intended.
     SpendsChange(StepOutput),
+    /// The proposal attempts to send a memo to a transparent recipient.
+    SendsMemoToTransparentRecipient,
```

Comment:
I'm surprised that this error variant is needed; the newly added paths at the high level shouldn't add any new error cases that have to be propagated out.